src/hooks/use-sound-wave-rendering.ts

```diff
@@ -99,123 +99,129 @@ const drawZoomAreaMarker = (props: IDrawHelperProps) => {
-    ctx.fillRect(x3Scaled, baseY - kIBeamWidth * 0.5, kLineWidth, kIBeamWidth);
-    // Draw (semi-opaque) background for the text
-    const kTextFontSize = 14;
-    const kTextTopPadding = 40;
-    const kBoxWidth = 150;
-    const kBoxHeight = kTextFontSize * 1.33;
-    ctx.fillStyle = "#ffffffb0";
-    ctx.fillRect((x1Scaled + x3Scaled) * 0.5 - 0.5 * kBoxWidth, baseY + 0.65 * kTextTopPadding, kBoxWidth, kBoxHeight);
-    // Label
-    ctx.fillStyle = "#892be2";
-    ctx.textAlign = "center";
-    ctx.font = `${kTextFontSize} Comfortaa, cursive`;
-    const wavelengthInMs = 1 / (pureToneFrequency || 1) * 1000;
-    ctx.fillText(`Wave length: ${wavelengthInMs.toFixed(2)}ms`, (x1Scaled + x3Scaled) * 0.5, baseY + kTextTopPadding);
-  }
-};
+//   const startIdx = currentDataPointIdx + markerPadding;
+
+//   let waveX1 = null; // X coord of the first sign change
+//   let waveX2 = null; // X coord of the second sign change
+//   let waveX3 = null; // X coord of the third sign change
+//   for (let i = startIdx; i < currentDataPointIdx + pointsCount * 0.5; i++) {
+//     const y1 = data[i - kSampleDiff];
+//     const y2 = data[i + kSampleDiff];
+//     if (y1 !== undefined && y2 !== undefined && Math.sign(y1) !== Math.sign(y2)) {
+//       if (waveX1 === null) {
+//         waveX1 = i;
+//         i += kSampleDiff * 2;
+//       } else if (waveX2 === null) {
+//         waveX2 = i;
+//         i += kSampleDiff * 2;
+//       } else {
+//         waveX3 = i;
+//         break;
+//       }
+//     }
+//   }
+
+//   if (waveX1 !== null && waveX3 !== null) {
+//     const kLineWidth = 3;
+//     const xScale = width / pointsCount;
+//     const minPointIdx = currentDataPointIdx - 0.5 * pointsCount; // currentDataPointIdx is always in the middle
+//     const x1Scaled = (waveX1 - minPointIdx) * xScale;
+//     const x3Scaled = (waveX3 - minPointIdx) * xScale;
+
+//     // Horizontal line
+//     ctx.fillStyle = "#892be2";
+//     ctx.fillRect(x1Scaled, baseY - 0.5 * kLineWidth, (x3Scaled - x1Scaled), kLineWidth);
+
+//     // I-beam
+//     const kIBeamWidth = 12;
+//     ctx.fillRect(x1Scaled - kLineWidth, baseY - kIBeamWidth * 0.5, kLineWidth, kIBeamWidth);
+//     ctx.fillRect(x3Scaled, baseY - kIBeamWidth * 0.5, kLineWidth, kIBeamWidth);
+
+//     // Draw (semi-opaque) background for the text
+//     const kTextFontSize = 14;
+//     const kTextTopPadding = 40;
+//     const kBoxWidth = 150;
+//     const kBoxHeight = kTextFontSize * 1.33;
+//     ctx.fillStyle = "#ffffffb0";
+//     ctx.fillRect((x1Scaled + x3Scaled) * 0.5 - 0.5 * kBoxWidth, baseY + 0.65 * kTextTopPadding, kBoxWidth, kBoxHeight);
+
+//     // Label
+//     ctx.fillStyle = "#892be2";
+//     ctx.textAlign = "center";
+//     ctx.font = `${kTextFontSize} Comfortaa, cursive`;
+//     const wavelengthInMs = 1 / (pureToneFrequency || 1) * 1000;
+//     ctx.fillText(`Wave length: ${wavelengthInMs.toFixed(2)}ms`, (x1Scaled + x3Scaled) * 0.5, baseY + kTextTopPadding);
+//   }
+// };
```

Comment:
This is a lot of commented code that we are leaving in.

Reply:
Just FYI, the Story Owner recommended commenting-out the code, over deleting it. But, in deference to team consensus, I've deleted it now.

Reply:
If there's a realistic chance of wanting to return to it, the other approach would be to leave it in but behind a runtime flag -- which I think Piotr also suggested as a possible alternative. That way it's guaranteed that the code actually compiles and it's easy to come back to it if necessary. That's what I did with the circles/symbols at the end of the plot traces in Juice, for instance.